stm32-modules/heater-shaker/include/hal/message_queue.hpp

```diff
@@ -19,10 +19,17 @@ concept MessageQueue = requires(MQ mq, MessageType mt, const MQ cmq,
     ->std::same_as<bool>;
     { mq.try_send(cmt) }
     ->std::same_as<bool>;
+    // Queues must have a receive that takes a message instance by ptr
```

Comment:
I'm still a little confused as to why we have both a receive and a try/receive + what the different use cases would be. By instinct, I would assume the message queue would only need a try receive.

Reply:
the difference is basically "do i want to do this task to do stuff other than wait for messages". for the usb queue, it has literally nothing to do other than wait for messages, so there's no point in it using `try_recv` and waking up every time the timeout pops - it will just loop back to the `try_recv` 100% of the time. but for something like the heater task, 
- it has to receive messages (and send them for that matter)
- it also has to run a control loop once every say 100ms
so there, doing a `try_recv` with a 100ms timeout is a better choice because the ultimate goal is that you fold in every single wake-up condition into a single syscall.